.github/workflows/canary-integration-test.yml

```diff
@@ -105,6 +105,27 @@ jobs:
         kubectl -n rook-ceph exec $toolbox -- rbd namespace create replicapool/radosNamespace
         kubectl -n rook-ceph exec $toolbox -- python3 /etc/ceph/create-external-cluster-resources.py --cephfs-filesystem-name myfs --rbd-data-pool-name replicapool --rados-namespace radosNamespace --cluster-name rookStorage --restricted-auth-permission true
 
+    - name: test the upgrade flag
+      run: |
+        toolbox=$(kubectl get pod -l app=rook-ceph-tools -n rook-ceph -o jsonpath='{.items[*].metadata.name}')
+        # print existing client auth
+        kubectl -n rook-ceph exec $toolbox -- ceph auth ls
+        # update the existing non-restricted client auth with the new ones
+        kubectl -n rook-ceph exec $toolbox -- python3 /etc/ceph/create-external-cluster-resources.py --upgrade
```

Comment:
Yes this upgrades the users to new permissions,
For example:
```
client.csi-cephfs-provisioner
	key: AQDEQgFiHqnxHRAAU+nEHwivIdWS2xK4PX65Zw==
	caps: [mgr] allow rw
	caps: [mon] allow r
	caps: [osd] allow rw tag cephfs metadata=*
```
```
client.csi-cephfs-provisioner
	key: AQDEQgFiHqnxHRAAU+nEHwivIdWS2xK4PX65Zw==
	caps: [mgr] allow rw
	caps: [mon] allow r, allow command 'osd blocklist'
	caps: [osd] allow rw tag cephfs metadata=*
```

This is because the users created by rook doesn't have 'osd blocklist', but the updated script has.

Reply:
My bad I was looking at the restricted user `client.csi-cephfs-provisioner-rookStorage-myfs`.